Feat: restyled components, reworked spiderfication

john-salman opened the pull request:
- Reworked the sidebar to make styling responsive
- Added icons for buttons
- Added a summary for the search result entries
- Made table layout fullscreen
- Added concentric circle layout for spiderfied icons 
- Added missing legend entries

webb-ben commented:
> is there a way to get a deploy preview? maybe i am missing something obvious but i dont see the preview link

I think its because John in contributing from his fork. Requires approval to run otherwise

C-Loftus commented:
> I think its because John in contributing from his fork. Requires approval to run otherwise

If that is the case I think it might make sense to do the development directly in the repo instead of in a fork. I know forking is the best practice but realistically in a small repo like this one that has little visibility I think it is fine and would make iterating easier

webb-ben commented:
Lets review merge this and then @john-salman can you continue contributing but from a branch?

john-salman commented:
@C-Loftus @webb-ben Works for me

webb-ben commented on src/app/App.tsx:
```diff
@@ -15,21 +19,61 @@ type Props = {
+                     min-w-[300px] sm:min-w-[400px]
+                     max-w-[300px] sm:max-w-[400px]
+                     h-[88vh] lg:h-full
+                     flex overflow-hidden bg-primary
```
Not sure this should be turquoise. White might be okay

C-Loftus replied:
agree with this. something slightly opaque or not a solid color might look nice as well fwiw

webb-ben commented on src/app/components/Map/tools/Legend.tsx:
```diff
@@ -0,0 +1,140 @@
+                const color = getLayerColor(sublayer.id);
+
+                return (
+                    <div
```
Can this be a table or something?

webb-ben commented on src/app/features/MapTools/index.tsx:
```diff
@@ -92,17 +103,43 @@ export const MapTools: React.FC = () => {
+                        }
+                        custom={
+                            <div className="-mt-2">
+                                Dataset Clusters
```
Dont know if we need to include cluster in the legend

C-Loftus reviewed (commented):
Few thoughts:

Would be preferred if the table was entirely on screen with margins rather than going off screen. As a user, I was also not entirely clear what the table signified until I clicked around. Lets think about ways to make the UX more clear,. Perhaps the table tab shouldnt show up unless data is present?

![image](https://github.com/user-attachments/assets/c739a2d9-e7e2-418c-ae99-55b04ef522d2)

Agree with ben on colorscheme. Maybe we can just have the search be an opaque pop up over the map so we don't always take up the sidebar?

 Also red on a black base map seems a bit overly high contrast imo. I wonder if there is another colorscheme we can use? Do we want to allow a different base map color? It seems like black isn't as common and at least to me, sometimes signifies the map is still loading or something. All for light/dark theme / togglable basemaps though.

Pressing the settings button should probably close any settings if they are open

![image](https://github.com/user-attachments/assets/9643e3bf-3eef-4be8-a622-00ab3aed5b7b)

john-salman commented:
Pushed some more changes:
- Addressed feedback from Justin
- Changed from CGS turqouise to white for sidebar
- Removed clusters from legend
- Added additional padding to improve legibility in legend
- Added summary to sidebar after selection
- Retaining current layer color-scheme and basemap, will revisit later
- Tab opacity changes are visible with white bg, retaining current functionality
- Settings button closes any open tools
- Retaining current table layout until we determine what columns are needed

john-salman commented:
I'll also be adjusting the label position for huc boundaries and adding a help splash shortly.

webb-ben commented:
Looks good! Because its a UI, I am going to continue nitpicking :) but these are only nitpicks

- When you hover a huc2 it sometimes displays the label multiple times. Not sure if thats what you mean with your previous comment John.
- When you click on a huc2, its impossible to then tell what mainstems are inside that huc or not.
- Related to the previous one, it might be worth only showing the "Large" mainstems. I find the initial load to be a bit overwhelming
- Should the search bar be collapsable? I am tempted to have it above the map and table tabs?
- Feels a bit redundant to have this
![image](https://github.com/user-attachments/assets/b9ea80eb-341f-4b05-888c-6d930ccd019f)

C-Loftus reviewed (commented):
Great work John! This looks really nice. Here are a few thoughts

- instead of having the search bar be collapsible, I think we may could have the search input box be always present but have the resulting datasests be collapsible. Or perhaps just make the search bar automatically be opened.
    - In the search box lets put some text regarding what the user can search for, or make the search more descriptive . 
- i am wondering if there is a way to make it more clear that the table only shows up once a dataset is selected. I feel like a more descriptive title than `Table` may make that easier to intuit
~~In the table the url appears to sometimes be encoded wrong, note the paragraph character in the url. this might be more suited for a future PR~~

https://waterservices.usgs.gov/nwis/iv/?sites=USGS:01538000%C2%B6meterCd=63160&format=rdb

![image](https://github.com/user-attachments/assets/81317520-1e98-458e-841c-910b0df80e7e)
![image](https://github.com/user-attachments/assets/3fa99b3f-5a99-468f-b4a0-3178c72607c2)

webb-ben commented:
> In the table the url appears to sometimes be encoded wrong, note the paragraph character in the url.
> * this might be more suited for a future PR

This is something wrong with the graph, that should be fixed once we switch to RENCI sparql

john-salman commented:
Thanks again for the feedback yall!

- Combining the search bar positioning feedback:
    - Maybe lets put the text input above the tabs, show results in a collapsible titled 'Search Results'?
    - I think the best spot for attribution would be at the top of the Search Bar, was thinking:
        - (Geoconnex/IoW icon) | Application Name | Help Icon for reshowing the help splash (positioned to the right) 
            - Any input on these pieces? May be a question for Kyle/USGS
    - Change 'Table' tab -> 'Data' | 'Results' (Results feels redundant)
- The huc labels are drawn letting mapbox do most of the work, planning to switch this to a point feature-collection geojson representing the centers
-  I hide the huc boundaries past a certain zoom level, want me to keep those around all the time or hide at a closer zoom?
- The summary shows even when making a selection via the map as well and is retained even if you hover over another search result
    - Should we add any additional info or would it make sense to reduce info in the search result summary to add value?
- As far as what to search for or how to use the search, that may make sense to keep that in the help splash. I'll think through that once I get that ironed out
